backend/src/controllers/authController.ts

```diff
@@ -21,7 +21,13 @@ class AuthController {
             });
 
             await newUser.save();
-            res.status(201).json({ message: "User registered successfully" });
+            const token = jwt.sign({ _id: newUser._id }, process.env.JWT_SECRET!, { expiresIn: '7d' });
+
+            res.status(201).json({
+                message: "User registered successfully",
+                user: newUser,
+                token
```

Comment:
On utilise les cookies plutôt pour ça :

```suggestion
            res.status(201).json({
                message: "User registered successfully",
                user: newUser,
                token
             })
                
            res.cookie("token", token, {
                httpOnly: true,
                secure: process.env.NODE_ENV === "production",
                sameSite: "strict",
                maxAge: 7 * 24 * 60 * 60 * 1000,
            })
```

Tu peux récupérer dans la response les cookies pour avoir le token si besoin.

Est-ce que tu peux créer une fonction private dans ce controller qui réunie ce code, que tu utilises dans la fonction register et login stp ?